textacy/fileio/utils.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+import bz2
+import gzip
+import io
+from itertools import tee, starmap
+try:
+    import lzma
+except ImportError:
+    pass  # Py2 doesn't include lzma in its stdlib
+import os
+import zipfile
+
+from cytoolz.itertoolz import cons, pluck
+
+from textacy import compat
+
+
+def open_sesame(filepath, mode='rt',
+                encoding=None, errors=None, newline=None):
+    """
+    Open file ``filepath``. Compression (if any) is inferred from the file
+    extension ('.gz', '.bz2', or '.xz') and handled automatically; '~', '.',
+    and/or '..' in paths are automatically expanded; if writing to a directory
+    that doesn't exist, all intermediate directories are created automatically,
```

Comment:
Hmm... maybe this should be an option?  Like, what if someone has a typo in the path.  Then they have to delete all the stuff that was accidentally created.